### PR TITLE
Created linkify.

### PR DESCRIPTION
**Pull request by darkowlzz**

### CODE REVIEW

File: waartaa/client/chat.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+const REGEX = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/g;
```
Comment (rtnpro) [inline waartaa/client/chat.js:1]: This REGEX is not precise, it's so greedy.

This `/(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?/` is a cleaner one, but very limited. Trying to improve on it.

**Comment by rtnpro** [comment]
@darkowlzz try this regex `/((?:https|http|ftp)?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})(:\d+)?\/?(\?[\w!\*'\(\);:@&=\+\$,\/\?%\[\]]*)?(#[\w\-]*)?/`

**Comment by rtnpro** [comment]
@darkowlzz 

I tried the above code with the following diff:

```
diff --git a/waartaa/client/chat.js b/waartaa/client/chat.js
index 6db028e498d8ef9e910248c4211e3e124e050538..c317ac19bbea3d6f8be61f3e7f44d4b2a32ce19e 100644
--- a/waartaa/client/chat.js
+++ b/waartaa/client/chat.js
@@ -1,4 +1,4 @@
-const REGEX = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/g;
+const REGEX =/((?:https|http|ftp)?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})(:\d+)?\/?(\?[\w!\*'\(\);:@&=\+\$,\/\?%\[\]]*)?(#[\w\-]*)?/g;

 updateHeight = function () {
   var body_height = $('body').height();
@@ -454,8 +454,11 @@ Meteor.subscribe("client", Meteor.user() && Meteor.user().username);

 Handlebars.registerHelper("linkify", function (message) {
  return new Handlebars.SafeString(
-   message.replace(REGEX, function(match) {
-     return "<a target='_blank' href='" + match + "'>" + match + "</a>";
+   message.replace(REGEX, function(match, p1) {
+      var _match = match;
+      if (!p1)
+        _match = 'http://' + match;
+      return "<a target='_blank' href='" + _match + "'>" + match + "</a>";
    })
  );
 });
```

Things look good. Can you update the PR so that I can merge this?

**Comment by darkowlzz** [comment]
I guess we should land this first and then improve the REGEX in another issue and use the above enhancement. The REGEX you made above has some issues, long links like some MDN links are half linkified.

**Comment by rtnpro** [comment]
@darkowlzz cool :) I will merge it then.

**Pull request merged**